src/utilities/paramValidation.ts

```diff
@@ -1,21 +1,30 @@
-import { ResolvedRoute, RouteMatchRule } from '@/types'
-import { getParamValue, getParamValuesFromUrl } from '@/utilities'
+import { Param, ResolvedRoute, RouteMatchRule } from '@/types'
+import { createMaybeRelativeUrl, getParamValue, getParamValuesFromUrl } from '@/utilities'
 
 export const routeParamsAreValid: RouteMatchRule = (route, url) => {
   try {
-    getRouteParams(route, url)
+    getRouteParamValues(route, url)
   } catch {
     return false
   }
 
   return true
 }
 
-export const getRouteParams = (route: ResolvedRoute, url: string): Record<string, unknown> => {
+export const getRouteParamValues = (route: ResolvedRoute, url: string): Record<string, unknown> => {
+  const { pathname, search } = createMaybeRelativeUrl(url)
+
+  return {
+    ...getRouteParams(route.pathParams, route.path, pathname),
+    ...getRouteParams(route.queryParams, route.query, search),
+  }
+}
```

Comment:
I left this function returning the combination of both, but this will need to be reworked. We can't just spread the values together, currently I believe query params of the same name as path would just override

Reply:
https://github.com/kitbagjs/router/issues/93

Reply:
great idea! https://github.com/kitbagjs/router/pull/92/files#diff-8875ff68754a59b04f6ca606e3f5d6a54351ee99c43a568ad292d4905e6d3c3cR105